src/js/index.js

```diff
@@ -37,91 +37,70 @@ $(document).ready(function () {
+      if (currentItem.location.country === 'gb') {
+        currentItem.location.country = 'uk';
+      }
+      if (!accumulator[currentItem.location.country]) {
+        accumulator[currentItem.location.country] = [];
+      }
+      accumulator[currentItem.location.country].push(currentItem);
+      return accumulator;
+    }, {});
 
-    if (x > y) {
-      return 1;
-    }
-    if (x < y) {
-      return -1;
-    }
-    return 0;
+    // sort by city
+    Object.keys(jobs).map(function (currentItem) {
+      jobs[currentItem].sort(sortCity);
+    });
+
+    // combine country arrays
+    var conbinedJobs = Object.keys(jobs).sort().reduce(function (accumulator, currentItem) {
+      return accumulator.concat(jobs[currentItem]);
+    }, []);
+
+    return conbinedJobs;
   }
 
   if ($('ul#jobs-board').length) {
-    (function () {
 
-      var url = "https://api.smartrecruiters.com/v1/companies/WiproDigital/postings?";
-      // get jobs that are Buildit not WD
-      var getBrand = 'custom_field.5880c55be4b0cfde272956ad=83455af9-c888-4221-9312-4750b5a09bf5';
+    var url = "https://api.smartrecruiters.com/v1/companies/WiproDigital/postings?";
+    // get jobs that are Buildit not WD
+    var getBrand = 'custom_field.5880c55be4b0cfde272956ad=83455af9-c888-4221-9312-4750b5a09bf5';
+
+    fetch(url + getBrand).then(function (response) {
+      if (!response.ok) {
+        throw Error(response.statusText);
+      }
+      return response.json();
+    })
+      .then(function (response) {
+        var wrapper = $('ul.opening-jobs');
+        var combinedData = processJobData(response.content);
+
+        wrapper.append(
+          combinedData
+            .map(tpl)
+            .join('')
+        );
 
-      fetch(url + getBrand).then(function (response) {
-        if (!response.ok) {
-          throw Error(response.statusText);
-        }
-        return response.json();
       })
-        .then(function (response) {
-          var data = response.content;
-          var wrapper = $('ul.opening-jobs');
-
-          data = data.reduce(divideByCountry, {});
-
-          var countries = Object.keys(data);
-          formattedData = countries
-            .map(function(country, index) {
-              data[country].sort(sortByCity);
-              if (country === 'gb') {
-                return 'uk';
-              }
-              return country;
-            })
-            .sort(sortCountry)
-            .map(function(country, index) {
-              if (country === 'uk') {
-                return data['gb'];
-              }
-              return data[country];
-            });
-          formattedData = [].concat.apply([], formattedData);
-          wrapper.append(
-            formattedData
-              .map(tpl)
-              .join('')
-          );
-        })
-        .catch(function (error) {
-          console.log(error);
-        });
-    })();
+      .catch(function (error) {
+        console.log(error);
```

Comment:
Is there a plan to put an error message or something else for the user to see if this happens? Right now all they see is the career page with just "...still interested?" but nothing else.